src/resolvers/Mutation/createChat.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  ORGANIZATION_NOT_FOUND_ERROR,
+  USER_NOT_FOUND_ERROR,
+} from "../../constants";
+import { errors, requestContext } from "../../libraries";
+import { Chat, Organization, User } from "../../models";
+import { cacheOrganizations } from "../../services/OrganizationCache/cacheOrganizations";
+import { findOrganizationsInCache } from "../../services/OrganizationCache/findOrganizationsInCache";
+import type { MutationResolvers } from "../../types/generatedGraphQLTypes";
+/**
+ * This function enables to create a chat.
+ * @param _parent - parent of current request
+ * @param args - payload provided with the request
+ * @param context - context of entire application
+ * @remarks The following checks are done:
+ * 1. If the user exists
+ * 2. If the organization exists
+ * @returns Created chat
+ */
+export const createChat: MutationResolvers["createChat"] = async (
+  _parent,
+  args,
+  context,
+) => {
+  let organization;
+  const usersInChat = [];
+  if (args.data.isGroup && args.data.organizationId) {
+    const organizationFoundInCache = await findOrganizationsInCache([
+      args.data.organizationId,
+    ]);
+
+    organization = organizationFoundInCache[0];
+
+    if (organizationFoundInCache.includes(null)) {
+      organization = await Organization.findOne({
+        _id: args.data.organizationId,
+      }).lean();
+      if (organization) await cacheOrganizations([organization]);
+    }
+
+    // Checks whether organization with _id === args.data.organizationId exists.
+    if (!organization) {
+      throw new errors.NotFoundError(
+        requestContext.translate(ORGANIZATION_NOT_FOUND_ERROR.MESSAGE),
+        ORGANIZATION_NOT_FOUND_ERROR.CODE,
+        ORGANIZATION_NOT_FOUND_ERROR.PARAM,
+      );
+    }
+  }
+
+  for await (const userId of args.data.userIds) {
```

Comment:
The loop uses for `await` to check user existence one by one. This results in sequential database queries, which can be slow.  Instead of checking each user individually, perform a bulk query to check all user IDs at once. This will significantly improve performance by reducing multiple database round-trips.

```javascript
const userExists = await User.exists({
  _id: { $in: args.data.userIds },
});

if (userExists.length !== args.data.userIds.length) {
  // Find which user ID(s) do not exist
  const missingUsers = args.data.userIds.filter(
    (id) => !userExists.includes(id)
  );
  throw new errors.NotFoundError(
    requestContext.translate(USER_NOT_FOUND_ERROR.MESSAGE),
    USER_NOT_FOUND_ERROR.CODE,
    { missingUsers }
  );
}
```